mysql-test/suite/encryption/t/percona_redo_default_key.test

```diff
@@ -0,0 +1,158 @@
+--source include/have_debug.inc
+
+let $MYSQLD_BASEDIR= `select @@basedir`;
+--mkdir $MYSQL_TMP_DIR/datadir1
+
+let $MYSQLD_DATADIR1 = $MYSQL_TMP_DIR/datadir1/data;
+--source include/shutdown_mysqld.inc
+
+--let BOOTSTRAP_SQL=$MYSQL_TMP_DIR/boot.sql
+--let KEYRING_FILE_PATH=$MYSQL_TMP_DIR/mysecret_keyring
+--let KEYRING_DATA="--keyring_file_data=$KEYRING_FILE_PATH"
+--let BOOTSTRAP_LOG=$MYSQL_TMP_DIR/bootstrap.log
+
+--echo # create bootstrap file
+write_file $BOOTSTRAP_SQL;
+CREATE DATABASE test;
+EOF
+
+--let BOOTSTRAP_CMD = $MYSQLD_CMD --initialize-insecure --basedir=$MYSQLD_BASEDIR --datadir=$MYSQLD_DATADIR1 --init-file=$BOOTSTRAP_SQL $KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD $KEYRING_DATA --innodb_redo_log_encrypt=KEYRING_KEY > $BOOTSTRAP_LOG 2>&1
+
+--exec $BOOTSTRAP_CMD
+
+--remove_file $BOOTSTRAP_LOG
+--remove_file $MYSQLD_DATADIR1/ib_logfile0
+--remove_file $MYSQLD_DATADIR1/ib_logfile1
+
+--let $test_error_log=$MYSQL_TMP_DIR/error_log.err
+
+# We crash before percona_redo:0 gets rotated to percona_redo-UUID:1 (it gets rotated when UUID becomes available)
+
+--error 137
+--exec $MYSQLD_CMD --loose-skip-log-bin --skip-log-slave-updates --datadir=$MYSQLD_DATADIR1 $KEYRING_DATA $KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD --log-error=$test_error_log --innodb_redo_log_encrypt=KEYRING_KEY --debug=+d,crash_before_redo_key_is_rotated
+
+# Here we check that server discovers that redo log was encrypted with percona_redo:0 (default encryption key for redo log file)
+# and is able to decrypt it.
+
+--let $restart_parameters = "restart: --loose-skip-log-bin --skip-log-slave-updates --datadir=$MYSQLD_DATADIR1 $KEYRING_DATA $KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD --log-error=$test_error_log --innodb_redo_log_encrypt=KEYRING_KEY"
+--replace_result $MYSQLD_DATADIR1 MYSQLD_DATADIR1 $MYSQL_TMP_DIR MYSQL_TMP_DIR $KEYRING_PLUGIN_OPT KEYRING_PLUGIN_OPT $KEYRING_PLUGIN_EARLY_LOAD KEYRING_PLUGIN_EARLY_LOAD
+--source include/start_mysqld.inc
+
+--source include/shutdown_mysqld.inc
+
+--remove_file $MYSQLD_DATADIR1/ib_logfile0
+--remove_file $MYSQLD_DATADIR1/ib_logfile1
+
+# We have removed redo log files. The server is no longer able to read percona_redo key's UUID from the redo log.
+# It should reuse the percona_redo:0 i.e. the default percona redo key and next generate and switch to using
+# new key - percona_redo-UUID:1 - when UUID becomes available.
```

Comment:
Is this a new key at this point? (uuid is unchanged, encryption was already turned on during bootstrap)

Reply:
You are right, this key exists because of the bootstrap. I have removed this part of the test, because it is redundant in this case.